src/base58.rs

```diff
@@ -0,0 +1,441 @@
+// Implementation is described here: https://tools.ietf.org/id/draft-msporny-base58-01.html
+// Part of the implementation below took inspiration from
+// https://github.com/dotcypress/base58check and
+// https://github.com/debris/base58
+// https://github.com/bitcoin/bitcoin/blob/master/src/base58.cpp
+// However, I tried not to look at the code so much, and instead tried to derive
+// some things myself.
+
+use crypto::{digest::Digest, sha2};
+
+const ALPHABET: &'static [u8; 58] = b"123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
```

Comment:
Base58 is weird, I thought you missed the characters: `0`, `I`, `O`, `l`, however that's the spec 😱

Reply:
I'm curious why did they chose these characters, instead of removing the last characters of the alphabet for example.

Maybe because these 4 are similar 🤷‍♀️

Reply:
Found this in docs/spec:

> It is different from Base64 in that the conversion alphabet has been carefully picked to work well in environments where a person, such as a developer or support technician, might need to visually confirm the information with low error rates

Reply:
And this:

> First, similar looking letters are omitted such as 0 (zero), O (capital o), I (capital i) and l (lower case L). Doing so eliminates the possibility of a human being mistaking similar characters for the wrong character. Second, the non-alphanumeric characters + (plus), = (equals), and / (slash) are omitted to make it possible to use Base58 values in all modern file systems and URL schemes without the need for further system-specific encoding schemes.

Reply:
Yea, that was a pretty smart thing Satoshi did haha [bech32](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki) will also be cool to implement

Reply:
Wow, interesting! I find it cool that most things related to crypto and blockchain have tons of new papers and research.